spec/models/your_details_form_spec.rb

```diff
@@ -95,4 +95,44 @@
       end
     end
   end
+
+  describe 'contribution_preference' do
+    context 'no contribution' do
+      let(:contribution_preference) { nil }
+
+      it 'should return minimum' do
+        expect(subject.contribution_preference).to eq('minimum')
+      end
+    end
+
+    context 'when the form is valid' do
+      context 'full contribution' do
+        let(:contribution_preference) { 'full' }
+
+        it 'should return full' do
+          expect(subject.contribution_preference).to eq('full')
+        end
+      end
+
+      context 'full contribution' do
+        let(:contribution_preference) { 'minimum' }
+
+        it 'should return minimum' do
+          expect(subject.contribution_preference).to eq('minimum')
+        end
+      end
+    end
+
+    context 'for salary below the minimum threshold' do
+      context 'minimum contribution' do
+        let(:salary) { 5_000 }
+        let(:contribution_preference) { 'minimum' }
+
+        it 'should return full' do
+          subject.valid?
+          expect(subject.contribution_preference).to eq('full')
+        end
+      end
+    end
```

Comment:
I would rename this test descriptions to not use should. We don't have yet a guideline for this, but I always tendo to follow from http://www.betterspecs.org/#should